docs/proposals/20230612-node-os-image-caching.md

```diff
@@ -116,74 +118,78 @@ Example of the enviornment variable being turned on:
+  - lastTransitionTime: "2023-06-20T17:43:39Z"
     status: "True"
-    type: ScaleSetRunning
+    type: NodeHealthy
+  latestModelApplied: true
+  ready: true
 ```
 
-Example AzureMachine yaml with the proposed additional needed timestamps in status:
+Example AzureMachine yaml with the important fields shown (want status.ready == true and each status: "True"):
 
 ```yaml
 apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
 kind: AzureMachine
 metadata:
   name: node-os-image-caching-machine
   namespace: default
+  creationTimestamp: "2023-06-20T17:38:39Z"
 status:
   conditions:
-  - lastTransitionTime: "2023-06-12T23:14:55Z"
+  - lastTransitionTime: "2023-06-20T17:41:16Z"
     status: "True"
-    type: ScaleSetModelUpdated
-  - lastTransitionTime: "2023-06-12T23:14:55Z"
+    type: Ready
+  - lastTransitionTime: "2023-06-20T17:41:16Z"
+    status: "True"
+    type: BootstrapSucceeded
+  - lastTransitionTime: "2023-06-20T17:40:16Z"
     status: "True"
-    type: ScaleSetRunning
+    type: DisksReady
+  - lastTransitionTime: "2023-06-20T17:40:16Z"
+    status: "True"
+    type: VMRunning
+  ready: true
 ```
 
 In terms of when to take a snapshot, a day is given as a general example which should be good for typical use but the specification of how often will be customizable as we know that certain operators have different strategies and use cases for how they’re running their services on our clusters.
 
-In terms of data model changes, AzureMachine and AzureMachinePool changes are likely and the changes we expect will be purely additive and nonbreaking. No removals should be required to the data model. For AzureMachineTemplate and AzureMachinePool we will add a new optional field under meta.annotations called nodeCachingModeInterval which will be enabled if present and it will map to an interval of 1 day by default.
+In terms of data model changes, AzureMachineTemplate and AzureMachinePool will be changed and the changes we expect will be purely additive and nonbreaking. No removals should be required to the data model. For AzureMachineTemplate and AzureMachinePool we will add a new optional field under spec.template.image called nodePrototyping which will be enabled if present and it have a required field under it called interval which will map to an interval of 1 day or 24 hours by default. For AzureMachinePool, we will also add an optional field under spec.strategy.rollingUpdate called prototypeAutomaticRollout which will be set to false by default since we don't want the replacement of a node OS image to automatically trigger a rolling update of all the nodes. This is because the new image will be functionally identical to the old one outside of name itself (since the old nodes will all have the update and security patch contents already present). If for some reason the operator wants to always trigger this rollout (maybe if they programatically use the image names themselves), then they can simply set this field to true.
```

Comment:
> For AzureMachinePool, we will also add an optional field under spec.strategy.rollingUpdate called prototypeAutomaticRollout which will be set to false by default since we don't want the replacement of a node OS image to automatically trigger a rolling update of all the nodes.

How will this behavior be achieved for AzureMachineTemplates?

Reply:
There isn't a great/clean way to do this, which is why we've moved AzureMachineTemplates out of scope